streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -717,7 +717,15 @@ private void runLoop() {
 
     private void enforceRebalance() {
         consumer.unsubscribe();
-        consumer.subscribe(builder.sourceTopicPattern(), rebalanceListener);
+        subscribeConsumer();
+    }
+
+    private void subscribeConsumer() {
+        if (builder.usesPatternSubscription()) {
```

Comment:
This is the actual fix; if the user has not themselves added pattern source topics we will go back to using regular subscription (having disabled auto topic creation for the main consumer)